src/d3a/models/market/two_sided_pay_as_bid.py

```diff
@@ -68,14 +68,16 @@ def delete_bid(self, bid_or_id: Union[str, Bid]):
 
     def accept_bid(self, bid: Bid, energy: float = None,
                    seller: str = None, buyer: str = None, already_tracked: bool = False,
-                   price_drop: bool = True):
+                   price_drop: bool = True, trade_rate: float = None):
         market_bid = self.bids.pop(bid.id, None)
         if market_bid is None:
             raise BidNotFound("During accept bid: " + str(bid))
 
         seller = market_bid.seller if seller is None else seller
         buyer = market_bid.buyer if buyer is None else buyer
         energy = market_bid.energy if energy is None else energy
+        if trade_rate is None:
```

Comment:
Similar to the one-sided market case, an assert needs to be added to validate that the trade rate is never higher than the bid rate.